nilearn/_utils/niimg_conversions.py

```diff
@@ -237,66 +286,45 @@ def check_niimgs(niimgs, accept_3d=False, return_iterator=False):
 
     Notes
     -----
-    This function is the equivalent to check_niimg() for Niimg-like objects
+    This function is the equivalent to check_niimg_3d() for Niimg-like objects
     with a session level.
 
     Its application is idempotent.
     """
-    # Initialization:
-    # If given data is a list, we count the number of levels to check
-    # dimensionality and make a consistent error message.
-    depth = 0
-    first_img = niimgs
-    if accept_3d and (isinstance(first_img, string_types)
-                      or not isinstance(first_img, collections.Iterable)):
-        niimg = check_niimg(niimgs)
-        if len(niimg.shape) == 3:
-            niimg = new_img_like(niimg, niimg.get_data()[..., np.newaxis],
-                            niimg.get_affine())
-        return niimg
+    # Two types of input:
+    # * 4D nifti image
+    # * list of 3D niimgs
 
+    # We get rid of the 4D nifti image case, as it is the simplest case
     # Use hasattr() instead of isinstance to workaround a Python 2.6/2.7 bug
     # See http://bugs.python.org/issue7624
-    while hasattr(first_img, "__iter__") \
-            and not isinstance(first_img, string_types):
-        if hasattr(first_img, '__len__') and len(first_img) == 0:
-            raise TypeError('An empty object - %r - was passed instead of an '
-                            'image or a list of images' % niimgs)
-        first_img = next(iter(first_img))
-        depth += 1
-
-    # First image is supposed to be a path or a Niimg-like object
-    first_img = check_niimg(first_img)
-
-    # Check dimension and depth
-    shape = first_img.shape
-    dim = len(shape)
-
-    if (dim + depth) != 4:
-        # Detailed error message that tells exactly the user what
-        # was provided and what should have been provided.
-        raise TypeError("Data must be a 4D Niimg-like object. You provided a "
-                        "%s%dD image(s), of shape %s. "
-                        "See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg." % (
-                        'list of ' * depth, dim, shape))
-
-    # Now, we load data as we know its format
-    if dim == 4:
-        if return_iterator:
-            result = (_index_niimgs(first_img, i)
-                      for i in range(shape[3]))
-        else:
-            result = first_img
-    else:
+    if isinstance(niimgs, string_types) or not hasattr(niimgs, "__iter__"):
+        niimgs = load_niimg(niimgs)
+        shape = niimgs.shape
+        if len(shape) == 3:
+            raise TypeError("A 4D image is expected, but an image "
+                "with a shape of %s was given." % (shape, ))
```

Comment:
Not the slightest idea ;). I copy pasted another error message but tuples for single string replacement are all around the code.